_posts/2020-10-12-sparklyr-flint-0.2.0-released/sparklyr-flint-0.2.0.Rmd

```diff
@@ -0,0 +1,143 @@
+
+## OLS Regression
+
+You might be wondering whether the version of this functionality in Flint is more or less identical to `lm()` in R. Turns out it has much more to offer than `lm()` does. A OLS regression in Flint will compute useful metrics such as [Akaike information criterion](https://en.wikipedia.org/wiki/Akaike_information_criterion) and [Bayesian information criterion](https://en.wikipedia.org/wiki/Bayesian_information_criterion), both of which are useful more model selection purposes, and the calculations of both are parallelized by Flint to fully utilize computational power available in a Spark cluster. In addition, Flint supports ignoring regressors that are constant or nearly constant, which becomes useful when an intercept term is included. To see why this is the case, we need to briefly examine the goal of the OLS regression, which is to find some column vector of coefficients $\mathbf{\beta}$ that minimizes $\|\mathbf{y} - \mathbf{X} \mathbf{\beta}\|^2$, where $\mathbf{y}$ is the column vector of response variables, and $\mathbf{X}$ is a matrix consisting of columns of regressors plus an entire column of $1$s representing the intercept terms. The solution to this problem is $\mathbf{\beta} = (\mathbf{X}^\intercal\mathbf{X})^{-1}\mathbf{X}^\intercal\mathbf{y}$, assuming the Gram matrix $\mathbf{X}^\intercal\mathbf{X}$ is non-singular. However, if $\mathbf{X}$ contains a column of all $1$s of intercept terms, and another column of formed by a regressor that is constant (or so), then columns of $\mathbf{X}$ will be linearly dependent (or nearly so) and $\mathbf{X}^\intercal\mathbf{X}$ will be singular (or nearly so), which presents an issue computation-wise. However, if a regressor is constant, then it essentially plays the same role as the intercept terms do, so simply excluding such constant regressor in $\mathbf{X}$ solves the problem. Also, speaking of inverting the Gram matrix, readers remembering the concept of "condition number" from numerical analysis must be thinking to themsolves how computing $\mathbf{\beta} = (\mathbf{X}^\intercal\mathbf{X})^{-1}\mathbf{X}^\intercal\mathbf{y}$ could be numerically unstable if $\mathbf{X}^\intercal\mathbf{X}$ has a large condition number. This is why Flint also outputs the condition number of the Gram matrix in the OLS regression result, so that one can sanity-check the underlying quadratic minimization problem being solved is well-conditioned.
+
+So, to summarize, the OLS regression functionality implemented in Flint not only outputs the solution to the problem, but also calculates useful metrics that helps data scentists assess the sanity and predictive quality of the resulting model.
+
+To see OLS regression in action in `sparklyr.flint`, one can run the following example:
```

Comment:
```suggestion
To see OLS regression in action with `sparklyr.flint`, one can run the following example:
```

Reply:
Not wrong at all, just changing this one because you have "in action" right before it.